File: sdk/block.event.ts
import { Block } from "./block";
import { EventType, Network } from "./index";

export class BlockEvent {
  constructor(
    readonly type: EventType,
    readonly network: Network,
    readonly block: Block
  ) {}

  get blockHash() {
    return this.block.hash
  }

  get blockNumber() {
    return this.block.number
  }
}

File: python-sdk/src/forta_agent/block_event.py
from .event_type import EventType
from .network import Network
from .block import Block


class BlockEvent:
    def __init__(self, dict):
        typeVal = dict.get('type', "BLOCK")
        self.type = EventType[typeVal] if type(
            typeVal) == str else EventType(typeVal)
        networkVal = dict.get('network', "MAINNET")
        networkVal = int(networkVal) if type(
            networkVal) == str and networkVal.isdigit() else networkVal
        self.network = Network[networkVal] if type(
            networkVal) == str else Network(networkVal)
        self.block = Block(dict.get('block', {}))

    @property
    def block_hash(self):
        return self.block.hash

    @property
    def block_number(self):
        return self.block.number
